packages/tpetra/core/test/CrsGraph/CrsGraph_UnitTests_Swap.cpp

```diff
@@ -0,0 +1,390 @@
+
+double      errorTolSlack = 1e+1;
+std::string filedir;
+
+#define STD_TESTS(graph)                                                                                      \
+    {                                                                                                         \
+        auto   STCOMM   = graph.getComm();                                                                    \
+        auto   STMYGIDS = graph.getRowMap()->getNodeElementList();                                            \
+        size_t STMAX    = 0;                                                                                  \
+                                                                                                              \
+        for(size_t STR = 0; STR < graph.getNodeNumRows(); ++STR)                                              \
+        {                                                                                                     \
+            TEST_EQUALITY(graph.getNumEntriesInLocalRow(STR), graph.getNumEntriesInGlobalRow(STMYGIDS[STR])); \
+            STMAX = std::max(STMAX, graph.getNumEntriesInLocalRow(STR));                                      \
+        }                                                                                                     \
+        TEST_EQUALITY(graph.getNodeMaxNumRowEntries(), STMAX);                                                \
+        GST STGMAX;                                                                                           \
+        Teuchos::reduceAll<int, GST>(*STCOMM, Teuchos::REDUCE_MAX, STMAX, Teuchos::outArg(STGMAX));           \
+        TEST_EQUALITY(graph.getGlobalMaxNumRowEntries(), STGMAX);                                             \
+    }
+
+
+TEUCHOS_STATIC_SETUP()
+{
+    Teuchos::CommandLineProcessor& clp = Teuchos::UnitTestRepository::getCLP();
+    clp.setOption("filedir", &filedir, "Directory of expected input files.");
+    clp.addOutputSetupOptions(true);
+    clp.setOption("test-mpi",
+                  "test-serial",
+                  &Tpetra::TestingUtilities::testMpi,
+                  "Test MPI (if available) or force test of serial.  In a serial build,"
+                  " this option is ignored and a serial comm is always used.");
+    clp.setOption("error-tol-slack", &errorTolSlack, "Slack off of machine epsilon used to check test results");
+}
+// todo: update options so that this test won't run in serial (it's not set up for that currently)
+
+
+
+
+// comm           : a communicator with exactly 2 processes in it.
+// edges          : [ (u,v), ... ]
+// row_owners     : [ (u, pid), ... ]
+// gbl_num_columns: Max # of columns in the matrix-representation of the graph.
+//                  This should be >= the highest value of v from all edges (u,v) in edges.
+//                  Note: u and v are 0-indexed, so if the highest v is 11, then this should be 12.
+template<class LO, class GO, class Node, class comm_t>
```

Comment:
Why are you templating on `comm_t`?  It's always `Teuchos::Comm<int>`.  That's the common base class of `Teuchos::MpiComm<int>` and `Teuchos::SerialComm<int>`.